pennylane/operation.py

```diff
@@ -649,6 +649,16 @@ def id(self):
     def name(self, value):
         self._name = value
 
+    @property
+    def pauli_rep(self):
+        """The reduced pauli representation of an operator.
+        Will be None for most operators except for pauli ops."""
+        return self._pauli_rep
+
+    @pauli_rep.setter
+    def pauli_rep(self, rep):
+        self._pauli_rep = rep
+
```

Comment:
I wonder if we should add these to a mix-in class instead of the base class. Then only operators that could conceivably have a `pauli_rep` would define a `pauli_rep`.  

Then whenever accessing it, we would do something like `isinstance(op, Pauli)`.

Reply:
I think that's a pretty solid option, Im not sure what the pros / cons of that would be relative to this implementation. I guess for starters we wouldn't be inflating the Operator class with more attributes